### ISSUE TITLE
Deepin not launching programs

### ISSUE THREAD

**Issue by thegleich** [NixOS/nixpkgs#300256]
### Describe the bug
Running any app from Deepin launchpad, or from Dock does nothing. There is no error shown and nothing happens.

The only way I found to open Terminal is to create a new icon on Desktop and then opening it with a Neovim wrapper which puts me into a terminal session.

### Steps To Reproduce
```
xserver = {
  desktopManager = { deepin = { enable = true; }; };
};
```
### Expected behavior
I expect the application to start.

### Additional context
Started after updating to newest nixos-unstable release, was fine with a 3 weeks old version.

Running programs from the terminal is working fine.
There might be a related warning/error in `./cache/dde-launchpad/dde-launchpad.log`:
```
2024-03-30, 16:33:31.453 [Warning] [  ] The X11 connection broke (error 1). Did the X11 server die?
```

Only service I see as failing is deepin-accounts1-daemon-service
```
ožu 30 16:09:48 earth systemd[1]: Starting Accounts1 Service...
ožu 30 16:09:48 earth systemd[1]: Started Accounts1 Service.
ožu 30 16:09:48 earth dde-system-daemon[2006]: <warning> virtual.go:49: load version file failed, err:  open /usr/share/dde-daemon/supportVirsConf.ini: no such file or directory
ožu 30 16:09:48 earth dde-system-daemon[2006]: <info> virtual.go:39: support virtual :  [hvm bochs virt vmware kvm cloud invented]
ožu 30 16:09:48 earth daemon/dde-system-daemon[2006]: virtual.go:49: load version file failed, err:  open /usr/share/dde-daemon/supportVirsConf.ini: no such file or directory
ožu 30 16:09:48 earth daemon/dde-system-daemon[2006]: virtual.go:39: support virtual :  [hvm bochs virt vmware kvm cloud invented]
ožu 30 16:09:48 earth dde-system-daemon[2006]: <warning> main.go:73: name "org.deepin.dde.Daemon1" already has the owner
ožu 30 16:09:48 earth daemon/dde-system-daemon[2006]: main.go:73: name "org.deepin.dde.Daemon1" already has the owner
ožu 30 16:09:48 earth systemd[1]: deepin-accounts1-daemon.service: Main process exited, code=exited, status=1/FAILURE
ožu 30 16:09:48 earth systemd[1]: deepin-accounts1-daemon.service: Failed with result 'exit-code'.
```

### Notify maintainers

@wineee 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
➜  Desktop nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.23, NixOS, 24.05 (Uakari), 24.05.20240329.d8fe5e6`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/cb1gs888vfqxawvc65q1dk6jzbayh3wz-source`
```

**Comment by wineee** [comment]
I can't reproduce this issue, but I've encountered similar problems before, it should fixed in https://github.com/linuxdeepin/dde-application-manager/pull/165

`dde-application-manager` will launch the application through systemd，and systemd will search for executable files in `ExecSearchPath` instead of `PATH`. So `dde-application-manager` will listen to the systemd environment variable and synchronize `PATH` to `ExecSearchPath`

You should check the `systemd` environment variables and logs to confirm that the PATH is set correctly

**Comment by thegleich** [comment]
It seems like dde-application-manager is failing to start. It fails with a std::terminate call here: https://github.com/linuxdeepin/dde-application-manager/blob/02bebd4ef08e0122de390c8faa82bc81f55fbe4f/src/systemdsignaldispatcher.h#L38:

```
dde-application-manager
register object: "/org/desktopspec/ProcessGuesser1" "org.desktopspec.ProcessGuesser1"
register object: "/org/desktopspec/ApplicationManager1" "org.desktopspec.ApplicationManager1"
register object: "/org/desktopspec/ApplicationManager1/JobManager1" "org.desktopspec.JobManager1"
register object: "/org/desktopspec/ApplicationManager1/MimeManager1" "org.desktopspec.MimeManager1"
Process org.freedesktop.systemd1 exited with status 1
Aborted (core dumped)
```

Output from: ```journalctl --system --follow```
```
tra 01 17:14:43 earth dbus-daemon[2616]: [session uid=0 pid=2614] Activating service name='org.freedesktop.systemd1' requested by ':1.82' (uid=0 pid=14084 comm="/nix/store/n1fzi2vj7k6z3swl8qr90cldwp8y6gsm-dde-ap" label="kernel")
tra 01 17:14:43 earth dbus-daemon[2616]: [session uid=0 pid=2614] Activated service 'org.freedesktop.systemd1' failed: Process org.freedesktop.systemd1 exited with status 1
```

Call seems to originate here: https://github.com/linuxdeepin/dde-application-manager/blob/02bebd4ef08e0122de390c8faa82bc81f55fbe4f/src/systemdsignaldispatcher.h#L36

Trying to make the call manually doesn't seem to be an issue (assuming I didn't call it wrong):
```
dbus-send --dest=org.freedesktop.systemd1 \
          --print-reply \
          --type=method_call \
          /org/freedesktop/systemd1 \
          org.freedesktop.systemd1.Manager.Subscribe
method return time=1711984758.074849 sender=:1.1 -> destination=:1.130 serial=824 reply_serial=2
```

**Comment by thegleich** [comment]
Thanks for the assist. Managed to pinpoint the problem.

The problem was indeed in ```ExecSearchPath```, specifically I had paths such as ```/home/g//.nix-profile/bin``` (notice the double-slash).

This double-slash was a product of having an extra slash in _Home Manager_ configuration home directory. 
I specified the home directory (wrongly) as ```/home/user/``` instead of ```/home/user```. 
This caused ```PATH``` to contain paths with double-slashes which when passed to systemd through ```ExecSearchPath``` fail parsing and process never gets launched.
